src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -575,9 +581,9 @@ private static PropertyBag<ObjectTypeProperty> PopulatePropertyBag(SerializableO
 
                 var additionalPropertyValueType = additionalPropertiesProperty.Declaration.Type.Arguments[1];
                 JsonSerialization valueSerialization;
-                if (inheritedDictionarySchema is not null)
+                if (additionalPropertiesValueType is not null)
                 {
-                    valueSerialization = BuildJsonSerialization(inheritedDictionarySchema.ValueType, additionalPropertyValueType, false);
+                    valueSerialization = BuildJsonSerialization(additionalPropertiesValueType, additionalPropertyValueType, false);
```

Comment:
can we have a better name for this?
`additionalPropertiesValueType` and `additionalPropertyValueType` are quite confusing

Reply:
Ah I did not notice the other parameter name.
I will fix this.